widget/entry.go

```diff
@@ -1296,7 +1296,8 @@ func (r *entryContentRenderer) Objects() []fyne.CanvasObject {
 	defer r.content.entry.propertyLock.RUnlock()
 	// Objects are generated dynamically force selection rectangles to appear underneath the text
 	if r.content.entry.selecting {
-		return append(r.selection, r.objects...)
+		objs := append([]fyne.CanvasObject(nil), r.selection...)
+		return append(objs, r.objects...)
```

Comment:
The title indicates that the fix was to not append the objects, but this change looks like rearranged code that does the same thing?...

Reply:
Well, that's not true. `append` does not always generate a new slice. It only creates a new slice if the current slice does not have enough capacity, otherwise it writes it.

Reply:
I am a little unsure - if we are adding all of the `r.selection` and `r.objects` to a new slice is that not the same result as appending one to the other.
It will no longer be able to re-use the same memory as before, but without it there should not be phantom elements randomly added due to memory re-use...

Reply:
> I am a little unsure - if we are adding all of the r.selection and r.objects to a new slice is that not the same result as appending one to the other.

The underlying memory array is modified if we append one to the other.

> It will no longer be able to re-use the same memory as before, but without it there should not be phantom elements randomly added due to memory re-use...

Yes, that is idea and as far as I see, it is the easiest way. Other way is adding synchronization but maybe it would be less performant due to the timing with locks and unlocks.

Reply:
> > I am a little unsure - if we are adding all of the r.selection and r.objects to a new slice is that not the same result as appending one to the other.
> 
> 
> 
> The underlying memory array is modified if we append one to the other.
> 

I am not so sure that's correct. Appending something returns a new slice, not updates the old one, as far as I know. That's the whole reason why you do `slice = append(slice, stuff...)`.

Reply:
> Appending something returns a new slice, not updates the old one, as far as I know.

That's not the way `append` function works, you can try it:
https://play.golang.org/p/60hb6o4c_kW

> That's the whole reason why you do slice = append(slice, stuff...).

The main reason of doing it is to avoid resizing the slice (if it has enough capacity) or to do have a new one but just because you need more capacity.

Reply:
Interesting. You learn something new every day 🙂

Reply:
Yes, that was crazy for me too when I learned it two years ago 😅. But in that way, `append` is more efficient. Indeed to be more efficient this part:
```go
objs := append([]fyne.CanvasObject(nil), r.selection...)
return append(objs, r.objects...)
```
could be written like this:
```go
objs := make([]fyne.CanvasObject, 0, len(r.selection) + len(r.objects))
objs = append(objs, r.selection...)
return append(objs, r.objects...)
```
In that way, Objects() will not create slices with more capacity than we want. And `objs` slice is just the same first one not a new one.

Reply:
I would support that change, it seems clearer.

Reply:
p.s. mind blown! you may find this mistake in other places as I (and may be others) did not realise)

Reply:
> I would support that change, it seems clearer.

Got it. Will update.

> you may find this mistake in other places as I (and may be others) did not realise)

I see, it is not entirely a mistake if you don't mind the integrity of the slice, so we can just wait if anything fails then it should be fixed, if not I think that should be ok (memory re-used) :)

Reply:
done